agent/config/config.go

```diff
@@ -297,6 +304,25 @@ func environmentConfig() Config {
 		seelog.Warnf("Invalid format for \"ECS_NUM_IMAGES_DELETE_PER_CYCLE\", expected an integer. err %v", err)
 	}
 
+	instanceAttributesEnv := os.Getenv("ECS_INSTANCE_ATTRIBUTES")
+	attributeDecoder := json.NewDecoder(strings.NewReader(instanceAttributesEnv))
+	var instanceAttributes map[string]string
+
+	err = attributeDecoder.Decode(&instanceAttributes)
+	if err != io.EOF && err != nil {
+		msg := "Invalid format for ECS_INSTANCE_ATTRIBUTES. Expected a json hash"
+		seelog.Warnf(msg)
+		errs = append(errs, errors.New(msg))
+	}
+	for k := range(instanceAttributes) {
+		seelog.Debugf("Setting instance attribute %v: %v\n", k, instanceAttributes[k])
+	}
+
+	if len(errs) > 0 {
```

Comment:
We probably just need to be sure how we're handing errors returned from this function.  If we're logging them or trying to handle errors based on the type we'll need to unwrap the `MultiErr`.